wurst/objects/items/PlagueBomb.wurst

```diff
@@ -0,0 +1,163 @@
+package PlagueBomb
+
+// Standard library imports:
+import Assets
+import ItemObjEditing
+import ClosureEvents
+import ClosureTimers
+import ClosureForGroups
+import ClosuresForItems
+import ChannelAbilityPreset
+import StackNSplit
+
+// Local imports:
+import LocalObjectIDs
+import LocalAssets
+import UnitExtensions
+
+@configurable let PLAGUE_RANGE = 500
+
+constant PLAGUE_ABIL = compiletime(ABIL_ID_GEN.next())
+constant ABILITY_RADIUS = 350.
+constant ITEM_ABILITIES = commaList(PLAGUE_ABIL)
+
+constant TOOLTIP = "Plague Bomb"
+constant TOOLTIP_EXTENDED = "Rains down plague on target location. All meat and buildings in the area will become infected."
+constant DESCRIPTION = "Rains plague down on an area infecting all meat and buildings."
+
+constant MODEL = Buildings.barrelsUnit0
+constant ICON  = LocalIcons.btNPlagueBomb
+constant EFFECT = Abilities.plagueCloudCaster
+
+constant MISSILE_FLY_HEIGHT = 140.
+constant PROJECTILE_INTERVAL = 1/30
+constant EFFECT_DURATION = 8.
+
+class PlagueBomb extends ChannelAbilityPreset
+    construct(int newAbilityId)
+        super(newAbilityId, 1, true)
+        this.presetManaCost(lvl -> 0)
+        this.presetCooldown(lvl -> 0)
+        this.setAreaofEffect(1, ABILITY_RADIUS)
+        this.setHeroAbility(false)
+        this.setItemAbility(false)
+        this.presetTargetTypes(Targettype.POINT)
+        this.setName(TOOLTIP)
+        this.setBaseOrderID(1, "antimagicshell")
+        this.presetTooltipNormalExtended(lvl -> TOOLTIP_EXTENDED)
+        this.setIconNormal(ICON)
+        this.setIconTurnOff(ICON)
+        this.setIconResearch(ICON)
+        this.presetOption(Option.TARGETIMAGE, true)
+        this.presetAreaofEffect(lvl -> ABILITY_RADIUS)
+        this.setFollowThroughTime(1, 0)
+        this.setTargetsAllowed(1, "")
+
+
+@compiletime function createPlagueBombAbility()
+    new PlagueBomb(PLAGUE_ABIL)
+
+@compiletime function createPlagueBombItem()
+    new ItemDefinition(ITEM_PLAGUE_BOMB, ItemIds.PlagueBomb)
```

Comment:
The second parameter of ItemDefinition is origID, the id of the item you based off your new item, meaning your item will inherit the attribute of the item referenced by origID.

e.g: `new ItemDefinition(ITEM_ORACLE_POTION, 'pinv')`, 'pinv' being the id of potion of invisibilty.
I don't know if it's possible to define an item from scratch, but here's what [plague bomb definition](https://i.gyazo.com/64ddb861a1889b85f52676615554ba1d.png) look like in the editor.
Even though the definition in the editor seems blank, plague bomb still take into account the attribute you set below, but some fields are missing/undefined? For example, the item can't be pawned.

ItemIds.PlagueBomb is a generated ID you defined in LocalAssets, therefore it doesn't reference any existing item.

Reply:
You cannot generate one from scratch.